Pushbutton Handler

murgui opened the pull request:
Pushbutton handler, it is missing to test. However I would like to do it with the interface PCB as it is HW related and I guess it will be ready to go tomorrow. 

At the moment you can check the nice looking code ( ͡° ͜ʖ ͡°)

murgui committed:
Pushbutton Handler completely developed

NicolaiFransen reviewed (approved):
Looks nice. I like how you used the structpointer again.

murgui commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+
+void pushEventQualificationFSM(PushButton *pushbutton)
+{
+    switch(pushbutton->FSMState)
```
Missing break; in every case

murgui reviewed (commented):
To be changed

murgui commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+    return buttonStateValue;
+}
+
+int antiSlipHasBeenPressed(void)
```
substitute by speedRefDecreaseHasBeenPressed

murgui commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+    return !(pushbutton->functionPointer());
+}
+
+int regenerativeBrakingHasBeenPressed(void)
```
speedRefIncreaseHasBeenPressed

murgui commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+
+static struct pushButtonListTag
+{
+    PushButton regenerativeBraking;
```
referenceSource
referenceType
speedIncrease
speedDecrease

AitorTeran commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+    return buttonStateValue;
+}
+
+int torqueReferenceHasBeenPressed(void)
```
maybe better referenceSource?

murgui replied:
you got it👌

AitorTeran commented on Software/Development/SemesterProject/App/pushbutttonManager.c:
```diff
@@ -0,0 +1,114 @@
+    return buttonStateValue;
+}
+
+int cruiseControlHasBeenPressed(void)
```
maybe better referenceType?
Also, this function will have to change it's value under circunstances different from button being pressed. For example when the driver brakes while it's active, the reference type has to change from speed to torque. Same for when driver accelerates momentarely or other cases. 
How are this situations going to be handled? maybe better changing the name to referenceTypeHasChanged? 
Maybe this is done somewhere else, idk but its something to keep in mind. Also we have to take a look at priorities and what to do under every situation.

murgui replied:
reference type ✔️

priorities and doing this kind of situation analysis you describe is performed in reference handler. You are completely right that we definitely have to look very carefully into it.

AitorTeran reviewed (commented):
Looking good!

murgui committed:
v1.1

The changes include the suggestions from the pull request review, and also the change of the pushbuttons from regenBraking and antiSlip to the speed increase/decrease when the cruise control is enabled

murgui commented:
I just commited the changes. The pull request will be a draft until I get to test in HW. It is missing to schedule the module.

murgui committed:
Changed scheduling task

murgui committed:
Solved conflicts

murgui committed:
Solved conflicts

murgui committed:
Pushbutton Handler after testing, plus fixed digital input bug as it was unable to read positions higher than 16 bits
